### ISSUE TITLE
SPI for DAB Italia – all files are suddenly purged

### ISSUE THREAD

**Issue by andimik** [softsyst/qirx_issues#66]
See https://wetransfer.com/downloads/7f7b4f16f446e8b41df5eaace4011ccb20221227175811/a9485e9827c98f05e87c228cf0d1d14720221227175826/dd3600

The command
```
cat ./dabitalia_20221227.eti | odr-dabmod /dev/stdin -f  /dev/stdout -F u8 -a 0.8 > ./dabitalia_20221227.raw
```
will produce a 2.3 GB raw file.

84 files are in the SPI folder at the end but after around 4:58 Minutes all them are deleted 

![grafik](https://user-images.githubusercontent.com/24510556/209704384-08410e78-4fc6-457a-8162-eb7451eb4079.png)

(screenshot taken at 81 files)

Then the SPI folder got empty.
![grafik](https://user-images.githubusercontent.com/24510556/209704498-ef910522-a44e-405b-be3b-a4f5833e3bc9.png)

The key message from the log is:

```
MOTState Complete
Purging disk path C:\Users\andreas\AppData\Local/qirx4/SPI\E05009\
No xml Directory or no SId directories
Directory C:\Users\andreas\AppData\Local/qirx4/SPI\E05009\ purged.
```

[qirx.logqirx.log-2022-12-27--19-22-06.txt](https://github.com/softsyst/qirx_issues/files/10310143/qirx.logqirx.log-2022-12-27--19-22-06.txt)

**Comment by softsyst** [comment]
No idea what went wrong in that case. If there is no SId directory, then sth. went wrong. If it is urgent, I needed a raw file to reproduce it. If it can wait some time, I'll inspect it when I re-activated the Linux.
The purging message indicates that the MOT info was complete. The rest is speculation.

**Comment by andimik** [comment]
I'll test with another recoding in the next days.

**Comment by andimik** [comment]
I can reproduce it. 

Seems the issue is caused at the moment when Qirx wants to create the XML file from the `55376_55295.dat` file.

Here is a 6:49 min. recording from yesterday in raw u8 format (zipped).

![grafik](https://user-images.githubusercontent.com/24510556/209781528-b2845983-6159-4199-946a-014b0c3bce61.png)

https://wetransfer.com/downloads/6f62990a152e6191daf66a1bd4322a9b20221227215856/44a42812b3d4f2d806c13473033f91dc20221227215911/7b63a1

**Comment by andimik** [comment]
BTW: cQIRX crashed (= the cQIRX window closed without any error message) before converting the `*.dat` into `*.xml`, but leaves all the 81 files in directory `%LOCALAPPDATA%\qirxConsole\SPI\E05009`:

![grafik](https://user-images.githubusercontent.com/24510556/209792857-a87b5415-4ddf-449a-aa82-ae80b883de4e.png)

Remark: cQIRX does not allow loading rawfiles, so I've used

```
 odr-dabmod foo.eti -f  /dev/stdout -F u8 -a 0.8 | pv -L 4096000 | nc -l 127.0.0.1 1234
```
and sent it via rtl_tcp.

[qirx.logqirx.log-2022-12-28--10-12-35.txt](https://github.com/softsyst/qirx_issues/files/10313456/qirx.logqirx.log-2022-12-28--10-12-35.txt)

Some of the last lines before crashing are:

> Purging disk path C:\Users\andreas\AppData\Local/qirxConsole/SPI\E05009\
> No xml path for ECC:E0 EId:5009

This is slightly different from Qirx.

Please note the `/` as well. I don't know if Windows converts it to `\` internally.

**Comment by softsyst** [comment]
For the weird path: Looks wrong, but works. Should be repaired.

Thanks for the raw file. The culprit is the file extension of the binary file, from which the xml with the picture info is generated. It is .dat, in contrast to otherwise used extensions. Some use .bin, some .EIB, among others. I use these "allowed" extensions, .dat was not among them. As it's not part of the standard, anyone can use arbitrary names.

I'll try to use a method to get rid of the extensions in a reliable way.

Nevertheless, I repaired it in my 4.0.5 Beta, which features the Logo handling as discussed in the forum. It worked straightforwardly, and produced the following:

![Italy](https://user-images.githubusercontent.com/51386375/209819478-fb863279-f10e-4b0f-8c0b-f180a4499adf.jpg)

The logos appear - if not yet present - after about 4:30Minutes. The picture shows a run after the first retrieval, showing the pics immediately.

Thanks for your efforts!!

And wrt the file names, about which you wondered the other day in another thread: These are just the crc values of the pic files, thus the copying is done only when the pictures are not yet there. The names during the retrieval are two "Transport Ids", which might change from run to run, although the pictures remain the same. Some minor optimization.

**Comment by andimik** [comment]
>Thanks for your efforts!!

Well, thanks to you!

It seems the ensemble itself also has thumbnails. See the 4 files 

> 55376_55304.png
> 55376_55305.png
> 55376_55306.png
> 55376_55307.png

![55376_55307](https://user-images.githubusercontent.com/24510556/209826088-b99eb9e1-6853-4ca9-aa9c-386dcaed39a9.png)
![55376_55304](https://user-images.githubusercontent.com/24510556/209826075-1dc66a6d-5ef7-41a7-8239-c49819da4af2.png)
![55376_55305](https://user-images.githubusercontent.com/24510556/209826082-31fa4b2a-0220-4d24-88cf-73affb437bf5.png)
![55376_55306](https://user-images.githubusercontent.com/24510556/209826086-80b94e2b-cda7-475e-b0e9-af978715eb41.png)

**Comment by softsyst** [comment]
Fixed in 4.0.5
